test/kouta_index/filtered_toteutus_list_test.clj

```diff
@@ -128,6 +135,13 @@
         (let [oids (get-200-oids (str (toteutus-url) "&page=3&size=2&order-by=tila"))]
           (is (= [] oids)))))
 
+    (testing "Toteutus result contains hakutieto hakukohde organisaatio and tila"
+      (let [res (get-200 (str (toteutus-url toteutusOid1)))]
+        (is (= 1 (:totalCount res)))
+        (is (= 1 (-> res :result (first) :hakutiedot (count))))
+        (is (= 2 (-> res :result (first) :hakutiedot (first) :hakukohteet (count))))
+        (is (->>     res :result (first) :hakutiedot (first) :hakukohteet (every? #(contains-many? % :organisaatioOid :tila))))))
+
```

Comment:
Tuli tästä mieleen tajunnanvirtana sellainen ajatus että itse tykkään kirjoittaa let expressionit niin että ensin luodaan data ja sitten sitä käytetään.
Tähän tyyliin
```clojure
(testing "Toteutus result contains hakutieto hakukohde organisaatio and tila"
        (let [res (post-200 [toteutusOid1] "")
              hakutiedot-count (-> res :result (first) :hakutiedot (count))
              hakukohteet-count (-> res :result (first) :hakutiedot (first) :hakukohteet (count))
              has-organisaatio-and-tila (->> res :result (first) :hakutiedot (first) :hakukohteet (every? #(contains-many? % :organisaatioOid :tila)))]
          (is (= 1 (:totalCount res)))
          (is (= 1 hakutiedot-count))
          (is (= 2 hakukohteet-count))
          (is (true? has-organisaatio-and-tila))))
```
Tässä ajatuksena se että saa pidettyä erillään datan luonnin ja sen käytön, ja pystyy koodin lukijaa ohjattelemaan noilla muuttujien nimillä. Noh, nämä on osaltaan mielipidekysymyksiä, tuli vain mieleen kun itse tykkään niin kovasti käyttää muuttujia ja funktioita koodin "kommentoinnissa".

Reply:
Toi on hyvä, mä muokkaan näin